ls1intum/Artemis
# `Development`: Use custom built images for GitLab and Jenkins

### Checklist
#### General
- [ ] This is a small issue that I tested locally and was confirmed by another developer ~on a test server~. → Only local testing is possible here due to the creation of Docker containers for GitLab and Jenkins.

### Motivation and Context
Due to the increasing number of developers in Artemis' developer team using a Mac with an M1 SoC, all images used for creating Docker containers must support `arm64` architectures. Unfortunately, GitLab does not provide one.

As a logical consequence, we should use custom-built images for Artemis that support those architectures.

### Description
This PR introduces the usage of custom-built images for GitLab and Jenkins from <https://github.com/ls1intum/Artemis-Local-Setup-Docker> (once ls1intum/Artemis-Local-Setup-Docker#4 has been merged).

Not only do they now provide native `arm64` versions – they additionally contain all changes such as installing plugins, already. This means, the images can directly be used in `docker-compose` files, such as [`gitlab-jenkins-mysql.yml`-file](https://github.com/ls1intum/Artemis/blob/feature/gitlab-jenkins-arm/src/main/docker/gitlab-jenkins-mysql.yml), eliminating the need to patch and build the images locally.

Additionally, the tag for the MySQL image is also changed to `8.0.28-oracle`. By changing it from images based on `debian:buster-slim` to those that base on `oraclelinux:8-slim`, we now use images that support both `amd64` as well as `arm64` architectures.

### Steps for Testing
1. [Follow the steps for setting up a local development environment with GitLab and Jenkins in the setup guide](https://artemis-platform.readthedocs.io/en/latest/dev/setup/jenkins-gitlab/).

> Please note that if the PR ls1intum/Artemis-Local-Setup-Docker#4 is not merged yet, you will not have access to the new images. In this case, either fork the repository [Artemis-Local-Setup-Docker](https://github.com/ls1intum/Artemis-Local-Setup-Docker) and merge the branch `feature/gitlab-jenkins-images` into `main` to build your own images, or use the ones for [GitLab](https://github.com/users/hhontheim/packages/container/package/artemis-gitlab) and [Jenkins](https://github.com/users/hhontheim/packages/container/package/artemis-jenkins) from my fork. Please use the `latest` tag – others might not be compatible...
>
> :warning: You will have to change the image names in the e.g. [`gitlab-jenkins-mysql.yml`-file](https://github.com/ls1intum/Artemis/blob/feature/gitlab-jenkins-arm/src/main/docker/gitlab-jenkins-mysql.yml) for Artemis to reflect the correct repository: `ghcr.io/`~`ls1intum`~`/artemis-{gitlab|jenkins}:latest` → `ghcr.io/`**`hhontheim`**`/artemis-{gitlab|jenkins}:latest` or your GH username.

2. Verify that your local setup is working. This includes (but is not limited to):

- User management with GitLab and Jenkins
- Programming exercises are usable

### Review Progress

#### Code Review
#### Manual Tests
